packages/ui-badge/src/Badge/styles.js

```diff
@@ -0,0 +1,194 @@
+ * @param  {Object} themeOverride User provided overrides of the default theme mapping.
+ * @param  {Object} props the props of the component, the style is applied to
+ * @param  {Object} state the state of the component, the style is applied to
+ * @return {Object} The final style object, which will be used in the component
+ */
+const generateStyle = (theme, themeOverride, props, state) => {
+  const componentTheme = generateComponentTheme(theme, themeOverride)
+
+  const { type, variant, placement, standalone, pulse } = props
+
+  const top = placement.indexOf('top') > -1
+  const bottom = placement.indexOf('bottom') > -1
+  const start = placement.indexOf('start') > -1
+  const end = placement.indexOf('end') > -1
+  const center = placement.indexOf('center') > -1
+
+  const variantVariants = {
+    danger: { backgroundColor: componentTheme.colorDanger },
+    success: { backgroundColor: componentTheme.colorSuccess },
+    primary: { backgroundColor: componentTheme.colorPrimary }
+  }
+
+  const pulseBorderVariants = {
+    danger: { borderColor: componentTheme.colorDanger },
+    success: { borderColor: componentTheme.colorSuccess },
+    primary: { borderColor: componentTheme.colorPrimary }
+  }
+
+  const countPositions = {
+    ...(top && { top: `calc(-1 * ${componentTheme.countOffset})` }),
+    ...(bottom && { bottom: `calc(-1 * ${componentTheme.countOffset})` }),
+    ...(start && {
+      insetInlineStart: `calc(-1 * ${componentTheme.countOffset})`,
+      insetInlineEnd: 'auto'
+    }),
+    ...(end && {
+      insetInlineEnd: `calc(-1 * ${componentTheme.countOffset})`,
+      insetInlineStart: 'auto'
+    }),
+    ...(center && {
+      ...((end || start) && {
+        top: `calc(50% - (${componentTheme.size} / 2))`
+      }),
+      ...(start && {
+        insetInlineStart: 'auto',
+        insetInlineEnd: `calc(100% - ${componentTheme.countOffset})`
+      }),
+      ...(end && {
+        insetInlineEnd: 'auto',
+        insetInlineStart: `calc(100% - ${componentTheme.countOffset})`
+      })
+    })
+  }
+
+  const notificationPositions = {
+    ...(top && { top: componentTheme.notificationOffset }),
+    ...(bottom && { bottom: componentTheme.notificationOffset }),
+    ...(start && {
+      insetInlineStart: componentTheme.notificationOffset,
+      insetInlineEnd: 'auto'
+    }),
+    ...(end && {
+      insetInlineEnd: componentTheme.notificationOffset,
+      insetInlineStart: 'auto'
+    }),
+    ...(center && {
+      ...((end || start) && {
+        top: `calc(50% - (${componentTheme.sizeNotification} / 2))`
+      }),
+      ...(start && {
+        insetInlineStart: `calc(-1 * ${componentTheme.sizeNotification} / 2)`,
+        insetInlineEnd: 'auto'
+      }),
+      ...(end && {
+        insetInlineEnd: `calc(-1 * ${componentTheme.sizeNotification} / 2)`,
+        insetInlineStart: 'auto'
+      })
+    })
+  }
+
+  const notStandaloneTypeVariant = {
+    count: countPositions,
+    notification: notificationPositions
```

Comment:
The key here should not be `'&.notification'` and `'.count'`?

Reply:
count/notification are coming from the `type` property, so we can use it it like any other property variant map